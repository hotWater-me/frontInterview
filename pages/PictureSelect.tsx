import React, { useRef, useEffect, useState, useCallback } from "react";
import Styles from './index.less'

interface picturesType {
  id: string;
  name: string;
  url: string;
}

interface PictureSelectProps {
  pictures: picturesType[];
}
const { Fragment } = React;
export const PictureSelect: React.FC<PictureSelectProps> = (props) => {
  const { pictures } = props;
  return (
    <Fragment>
      {pictures.map((item: picturesType) => (
        <span className={Styles.imageMargin}>
          <span className={Styles.noCheck}></span>
          <img src={item.url} />
        </span>
      ))}
    </Fragment>
  );
};
