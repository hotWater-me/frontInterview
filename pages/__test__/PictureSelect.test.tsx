import React from "react";
import { shallow } from "enzyme";
import { PictureSelect } from "../PictureSelect";

const pictures = [
  {
    id: "1",
    name: "foo",
    url:
      "https://gw.alipayobjects.com/mdn/rms_d212b7/afts/img/A*LlfeSa8N0WgAAAAAAAAAAABkARQnAQ",
  },
  {
    id: "2",
    name: "foo",
    url:
      "https://gw.alipayobjects.com/mdn/rms_d212b7/afts/img/A*LlfeSa8N0WgAAAAAAAAAAABkARQnAQ",
  },
  {
    id: "3",
    name: "foo",
    url:
      "https://gw.alipayobjects.com/mdn/rms_d212b7/afts/img/A*LlfeSa8N0WgAAAAAAAAAAABkARQnAQ",
  },
];

const props = {
  value: [],
  setValue: jest.fn(),
  pictures,
};

const setup = () => {
  const wrapper = shallow(<PictureSelect {...props} />);
  return {
    wrapper,
    props,
  };
};

describe("PictureSelect", () => {
  it("renders correctly", () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it("click every Img", () => {
    const { wrapper, props } = setup();
    wrapper.find(".noCheck").at(0).simulate("click");
    expect(props.setValue).toHaveBeenCalled();
  });

  it("All check Img", () => {
    const { wrapper } = setup();
  console.log(wrapper.prop('value'))
  });
});
