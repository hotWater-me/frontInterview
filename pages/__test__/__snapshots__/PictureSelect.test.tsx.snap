// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PictureSelect renders correctly 1`] = `
<Fragment>
  <div
    onClick={[Function]}
  >
    <span
      className="allChecked"
    />
    <span
      className="allcheckText"
    >
      以选中
      0
      个文件
    </span>
  </div>
  <div
    className="margin16"
  >
    <span
      className="imageMargin"
      key="1"
    >
      <span
        className="noCheck"
        onClick={[Function]}
      />
      <img
        src="https://gw.alipayobjects.com/mdn/rms_d212b7/afts/img/A*LlfeSa8N0WgAAAAAAAAAAABkARQnAQ"
      />
      <br />
      <span
        className="wrap"
      >
        https://gw.alipayobjects.com/mdn/rms_d212b7/afts/img/A*LlfeSa8N0WgAAAAAAAAAAABkARQnAQ
      </span>
    </span>
    <span
      className="imageMargin"
      key="2"
    >
      <span
        className="noCheck"
        onClick={[Function]}
      />
      <img
        src="https://gw.alipayobjects.com/mdn/rms_d212b7/afts/img/A*LlfeSa8N0WgAAAAAAAAAAABkARQnAQ"
      />
      <br />
      <span
        className="wrap"
      >
        https://gw.alipayobjects.com/mdn/rms_d212b7/afts/img/A*LlfeSa8N0WgAAAAAAAAAAABkARQnAQ
      </span>
    </span>
    <span
      className="imageMargin"
      key="3"
    >
      <span
        className="noCheck"
        onClick={[Function]}
      />
      <img
        src="https://gw.alipayobjects.com/mdn/rms_d212b7/afts/img/A*LlfeSa8N0WgAAAAAAAAAAABkARQnAQ"
      />
      <br />
      <span
        className="wrap"
      >
        https://gw.alipayobjects.com/mdn/rms_d212b7/afts/img/A*LlfeSa8N0WgAAAAAAAAAAABkARQnAQ
      </span>
    </span>
  </div>
</Fragment>
`;
